test/e2e/test-custom-cluster-name.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# Copyright 2024 The KubeStellar Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SRC_DIR="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd)"
+source "${SRC_DIR}/setup-shell.sh"
+
+set -x # echo so that users can understand what is happening
+set -e # exit on error
+
+CUSTOM_CLUSTER_NAME="test-custom-cluster"
+
+:
+: -------------------------------------------------------------------------
+: "Create kubeflex cluster with custom name"
+:
+if [ ! -z "$(bin/kflex init -c "${CUSTOM_CLUSTER_NAME}" 2>&1)" ]; then
+    echo "ERROR: Failed to create kubeflex cluster with custom name"
```

Comment:
Since the previous kubeflex hosting cluster still exists, this cluster conflicts on port 9080 - the CI e2e test shows:

```
Bind for 0.0.0.0:9080 failed: port is already allocated
```

Before trying to create the cluster with the new name it should use the kind command to delete the existing kind cluster so there is no conflict on that port.

Reply:
See this run with e2E test failing because of ports conflicts: https://github.com/kubestellar/kubeflex/actions/runs/17498172259/job/49802603324?pr=521